### PR TITLE
Add FreeSpec packages to extra-dev

### PR DESCRIPTION
**Pull request by lthms**
This is a draft PR to add the three packages in the `FreeSpec` repository in `extra-dev`.  I create this early on to discuss the Category of these packages. I haven’t found one that seems suitable in the list currently used in `released`. Does adding `CS/Program Logic` sounds fine to you?

### CODE REVIEW

**Comment by palmskog** [comment]
@lthms There is no strong meaning of categories currently, but I would really prefer if you stuck to the existing categories. If you want to add some arbitrary phrase like "program logic", you can always do that as a keyword.

File: extra-dev/packages/coq-freespec-core/coq-freespec-core.dev/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+maintainer: "Thomas Letan <thomas.letan@ssi.gouv.fr"
+version: "dev"
+
+homepage: "https://github.com/ANSSI-FR/FreeSpec"
+dev-repo: "git+https://github.com/ANSSI-FR/FreeSpec.git"
+bug-reports: "https://github.com/ANSSI-FR/FreeSpec.git/issues"
+doc: "https://ANSSI-FR.github.io/FreeSpec"
+license: "GPLv3"
```
Comment (palmskog) [inline extra-dev/packages/coq-freespec-core/coq-freespec-core.dev/opam:9]: Is this `GPL-3.0-only` or `GPL-3.0-or-later`? Same for other packages.

File: extra-dev/packages/coq-freespec-core/coq-freespec-core.dev/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+maintainer: "Thomas Letan <thomas.letan@ssi.gouv.fr"
+version: "dev"
```
Comment (palmskog) [inline extra-dev/packages/coq-freespec-core/coq-freespec-core.dev/opam:3]:
```suggestion
```
Same for other packages.

**Comment by palmskog** [comment]
Builds also seem to be failing:
```
#         coqc theories/Core/.Init.aux,theories/Core/Init.{glob,vo} (exit 1)
# (cd _build/default && /builds/coq/opam-coq-archive/opam-root-4.09.0-2.0.6-sandbox/4.09.0/bin/coqc -init-file build.v -R theories/Core FreeSpec.Core theories/Core/Init.v)
# Load of rcfile failed.
# Error: System error: "Cannot read rcfile: build.v"
```

**Comment by lthms** [comment]
I have fixed the `build.v` issue, but there is another issue I am having that I need more time for investigating. In the meantime, thanks for your suggestions!

**Comment by lthms** [comment]
Okay, so there are some progress here. It looks like FreeSpec is not compatible with the `dev` version of Coq, and I will have to investigate Coq changelog to understand why (it looks like it is related to a change of how instance declaration and `Generalize All Variables` work together…

Is it okay for `extra-dev` packages not to be comptable with the `dev` version of Coq?

**Comment by clarus** [comment]
I would say this is fine, as the `dev` version of Coq could change and break at anytime anyway.

**Comment by palmskog** [comment]
`extra-dev` is indeed open both for packages compatible with Coq `master` and other Coq versions. This is essentially the wild west version of `released`, where Git-based packages are side by side with regular archive-based packages.

**Comment by lthms** [comment]
I think this PR is ready; the necessary changes have been integrated to FreeSpec.

**Pull request merged**